File: tests/fixtures/extract/sample.kt
// Fixture: functions, methods, lambdas, comments, small bodies.

class Basket(private val prices: Map<String, Int>) {
    fun total(items: List<String>): Int {
        // sum known items, ignore unknown ones
        var sum = 0
        for (item in items) {
            val price = prices[item] ?: continue
            sum += price
        }
        return sum
    }
}

fun describe(items: List<String>): String {
    val cleaned = items.map { item ->
        val trimmed = item.trim()
        val lowered = trimmed.lowercase()
        lowered.replaceFirstChar { c -> c.uppercase() }
    }
    return cleaned.joinToString(separator = ", ")
}

fun tiny(): Int = 1

File: tests/fixtures/extract/sample.rs
//! Fixture: functions, methods, closures, nested units, comments, small bodies.

pub struct Counter {
    total: i64,
}

impl Counter {
    /// Adds a batch of values, skipping negatives.
    pub fn add_batch(&mut self, values: &[i64]) -> usize {
        // count how many were accepted
        let mut accepted = 0;
        for value in values {
            if *value >= 0 {
                self.total += value;
                accepted += 1;
            }
        }
        accepted
    }
}

pub fn summarize(values: Vec<i64>) -> (i64, i64) {
    let positives: Vec<i64> = values
        .iter()
        .filter(|v| **v > 0) // small closure, filtered out
        .cloned()
        .collect();
    let doubled: Vec<i64> = positives
        .iter()
        .map(|value| {
            let bumped = value + 1;
            let scaled = bumped * 2;
            scaled - 1
        })
        .collect();
    (positives.iter().sum(), doubled.iter().sum())
}

fn tiny() -> i32 {
    1
}
